pkg/util/network.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/cache"
+	"github.com/DataDog/datadog-agent/pkg/util/ec2"
+	"github.com/DataDog/datadog-agent/pkg/util/gce"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GetNetworkID retrieves the network_id which can be used to improve network
+// connection resolution. This can be configured or detected.  The
+// following sources will be queried:
+// * configuration
+// * GCE
+// * EC2
+func GetNetworkID() (string, error) {
```

Comment:
Will this be used anywhere else than the host metadata provider ? If no we should move it there with the rest of the metadata logic.

Reply:
I'm not sure what you're asking. I thought this *was* the host metadata provider.

I mentioned it briefly in the commit description, we're also going to be calling `GetNetworkID()` in the process agent so we can attach this to container payloads.  I'm not sure if that changes things.

Reply:
:+1:, just making sure this function belong in `pkg/util`.